cf-oscore/src/main/java/org/eclipse/californium/oscore/HashMapCtxDB.java

```diff
@@ -292,8 +293,10 @@ public synchronized void updateSeqByToken(Token token, Integer seq) {
 	 */
 	private static String normalizeServerUri(String uri) throws OSException {
 		String normalized = null;
+		int port = -1;
 
 		try {
+			port = (new URI(uri)).getPort();
```

Comment:
`new URI(String)` isn't that cheap. Maybe you adapt the snippet to create the URI once and reuse it?

Reply:
Yes, makes sense. I will try to do that.

Reply:
Now done in commit 1f50f113af4dca6c51ac99bdb61a149ed71c10bc